packages/injected/src/ariaSnapshot.ts

```diff
@@ -72,11 +83,11 @@ export function generateAriaTree(rootElement: Element, options?: { forAI?: boole
       return;
 
     const element = node as Element;
-    let isVisible = !roleUtils.isElementHiddenForAria(element);
-    if (options?.forAI)
-      isVisible = isVisible || isElementVisible(element);
-    if (!isVisible)
+    if (!isVisible(element, options)) {
+      // skip this element, but still process its children: https://github.com/w3c/aria/issues/1055
+      processElement(ariaNode, element, []);
```

Comment:
I am not sure how this makes sense. Also, you can't process children as if they are visible. And we probably only want this for non-aria visibility.

Reply:
> I am not sure how this makes sense

How so? As in, you're surprised how Chromium and this issue interpret the ARIA spec, or as in this code is weird?

> Also, you can't process children as if they are visible.

I think we can, because the `isVisible` method takes parent elements into account when determining visibility.

> we probably only want this for non-aria visibility.

It already does so, based on my reading of `isElementHiddenForAria`. I added a test in https://github.com/microsoft/playwright/pull/36316/commits/9775c4a4425eb060a90cd5a3acb53c1d33b3cddc to ensure that.